Userland/Libraries/LibC/stdlib.cpp

```diff
@@ -182,6 +184,141 @@ inline int generate_unique_filename(char* pattern, size_t suffix_length, Callbac
     return EEXIST;
 }
 
+static bool is_infinity_string(char* parse_ptr, char** endptr)
+{
+    if (is_either(parse_ptr, 0, 'i', 'I')) {
+        if (is_either(parse_ptr, 1, 'n', 'N')) {
+            if (is_either(parse_ptr, 2, 'f', 'F')) {
+                parse_ptr += 3;
+                if (is_either(parse_ptr, 0, 'i', 'I')) {
+                    if (is_either(parse_ptr, 1, 'n', 'N')) {
+                        if (is_either(parse_ptr, 2, 'i', 'I')) {
+                            if (is_either(parse_ptr, 3, 't', 'T')) {
+                                if (is_either(parse_ptr, 4, 'y', 'Y')) {
+                                    parse_ptr += 5;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (endptr)
+                    *endptr = parse_ptr;
+
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+static bool is_nan_string(char* parse_ptr, char** endptr)
+{
+    // FIXME: Actually parse (or at least skip) the (n-char-sequenceopt) part
+    if (is_either(parse_ptr, 0, 'n', 'N')) {
+        if (is_either(parse_ptr, 1, 'a', 'A')) {
+            if (is_either(parse_ptr, 2, 'n', 'N')) {
+                if (endptr)
+                    *endptr = parse_ptr + 3;
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+template<FloatingPoint T>
+static T strtofloatingpoint(char const* str, char** endptr)
+{
+    // First, they decompose the input string into three parts:
+    char* parse_ptr = const_cast<char*>(str);
+
+    // An initial, possibly empty, sequence of white-space characters (as specified by isspace())
+    strtons(parse_ptr, &parse_ptr);
+
+    // FIXME: This strlen can be very expensive if we are parsing the first value of a long string
+    //        we can also stop at the first white space or other characters which will never occur
+    //        in a floating point.
+    auto length = strlen(parse_ptr);
```

Comment:
This `strlen` can certainly cause a performance regression for any usage of strtod on a huge string. However I looked through the usages in the repo and those all look fine. 
However ports or future programs might use this so if anyone is super concerned we can add a special version which checks for `\0` instead of `end`

Reply:
The `strlen` is gone and performance of `strtod` and `strtof` is like it was before for the most common cases, and sped up for long cases (defenitely on long strings with the a small number at the beginning)